aslam_cv_cameras/src/camera.cc

```diff
@@ -59,37 +63,175 @@ void Camera::printParameters(std::ostream& out, const std::string& text) const {
   out << "  image (cols,rows): " << imageWidth() << ", " << imageHeight() << std::endl;
 }
 
-bool Camera::operator==(const Camera& other) const {
-  // \TODO(slynen) should we include the id and name here?
+bool Camera::isEqualCameraImpl(const Camera& other) const {
   return (this->intrinsics_ == other.intrinsics_) &&
          (this->line_delay_nanoseconds_ == other.line_delay_nanoseconds_) &&
          (this->image_width_ == other.image_width_) &&
          (this->image_height_ == other.image_height_);
 }
 
-Camera::Ptr Camera::loadFromYaml(const std::string& yaml_file) {
-  try {
-    YAML::Node doc = YAML::LoadFile(yaml_file.c_str());
-    return doc.as<aslam::Camera::Ptr>();
-  } catch (const std::exception& ex) {
-    LOG(ERROR) << "Failed to load Camera from file " << yaml_file << " with the error: \n"
-               << ex.what();
+bool Camera::loadFromYamlNodeImpl(const YAML::Node& yaml_node) {
+  if(!yaml_node.IsMap()) {
+    LOG(ERROR) << "Unable to parse the camera because the node is not a map.";
+    return false;
+  }
+
+  // Determine the distortion type. Start with no distortion.
+  const YAML::Node& distortion_config = yaml_node["distortion"];
+  if (distortion_config.IsDefined() && !distortion_config.IsNull()) {
+    if(!distortion_config.IsMap()) {
+      LOG(ERROR)
+          << "Unable to parse the camera because the distortion node is not a sequence.";
```

Comment:
Corrected error message. Can't move the check up by one since you have to first check it's a valid yaml node, otherwise any other operations are meaningless.